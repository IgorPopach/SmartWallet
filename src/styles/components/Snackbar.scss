.snackbar {
    position: fixed;
    width: 100%;
    top: 1rem;
    left: 0;
    max-height: 80%;
    overflow: hidden;
    padding: 1rem;
}

// Small devices (landscape phones, 576px and up)
@media (min-width: 576px) {
    .snackbar {
        width: 80%;
        left: 10%;
        max-height: 90%;
    }
}

// Medium devices (tablets, 768px and up)
@media (min-width: 768px) {
    .snackbar {
        max-height: 60%;
    }
}

// Large devices (desktops, 992px and up)
@media (min-width: 992px) {
    .snackbar {
        width: 50%;
        left: 25%;
        height: 560px;
    }
}

// Extra large devices (large desktops, 1200px and up)
@media (min-width: 1200px) {}