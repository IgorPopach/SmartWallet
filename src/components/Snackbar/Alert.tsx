import React, { useCallback } from 'react';

import { Message } from './../../store/snackbar/types';

import './../../styles/components/Alert';

interface Props {
    message: Message,
    onSubmit: (id: string) => void
}

const Alert: React.FC<Props> = ({ message, onSubmit }) => {

    const submitHandler = () => onSubmit(message.id)

    return (
        <div className={`alert alert-${message.type}`}>
            <p>
                <strong>{message.title}</strong>
            </p>
            <p>
                {message.text}
            </p>
            <button
                type="button"
                className="close"
                onClick={submitHandler}
            >
                <span aria-hidden="true">&times;</span>
            </button>
        </div>
    )
}

export default Alert;