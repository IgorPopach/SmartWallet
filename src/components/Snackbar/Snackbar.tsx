import React, { useCallback } from 'react';
import { Dispatch } from 'redux';
import { connect } from 'react-redux';

import './../../styles/components/Snackbar';
import { AppState } from '../../store';
import { Message } from '../../store/snackbar/types';
import { deleteMessage } from '../../store/snackbar/actions';
import Alert from './Alert';

interface StateProps {
    messages: Message[],
    visible: boolean
}

type OnSubmitCallback = (id: string) => void

interface DispatchProps {
    submitAction: OnSubmitCallback
}

type Props = StateProps & DispatchProps

const Snackbar: React.FC<Props> = ({ messages, submitAction, visible }) => {

    const cls = visible? 'snackbar-show' : 'snackbar-hide';

    return (
        <div className={`snackbar ${cls}`}>
            {messages.map((mess) => <Alert key={mess.id} message={mess} onSubmit={submitAction} />)}
        </div>
    )
}

const mapStateToProps = (state: AppState): StateProps => ({
    messages: state.snackbar.messages,
    visible: state.snackbar.visible
})

const mapDispatchToProps = (dispatch: Dispatch): DispatchProps => ({
    submitAction: (id: string) => dispatch(deleteMessage(id))
})

export default connect(mapStateToProps, mapDispatchToProps)(Snackbar);