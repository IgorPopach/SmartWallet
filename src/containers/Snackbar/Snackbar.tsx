import React from 'react';

import './../../styles/components/Snackbar.scss';
import { Message } from '../../store/snackbar/types';
import Alert from '../../components/Alert/Alert';

interface StateProps {
    messages: Message[];
}

interface DispatchProps {
    closeAction: (id: string) => void;
}

type Props = StateProps & DispatchProps;

const Snackbar: React.FC<Props> = ({ messages, closeAction }) => {
    const alerts = messages.map((mess) => {
        const closeCallback = () => closeAction(mess.id);
        return <Alert key={mess.id} message={mess} onClose={closeCallback} delay={15000} />;
    });
    return <div className={`snackbar`}>{alerts}</div>;
};

export default Snackbar;
