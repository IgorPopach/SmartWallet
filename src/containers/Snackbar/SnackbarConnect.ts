import { Dispatch } from 'redux';
import { connect } from 'react-redux';

import Snackbar from './Snackbar';
import { AppState } from '../../store';
import { Message } from '../../store/snackbar/types';
import { deleteMessage } from '../../store/snackbar/actions';

interface StateProps {
    messages: Message[];
}

interface DispatchProps {
    closeAction: (id: string) => void;
}

const mapStateToProps = (state: AppState): StateProps => ({
    messages: state.snackbar.messages,
});

const mapDispatchToProps = (dispatch: Dispatch): DispatchProps => ({
    closeAction: (id: string) => dispatch(deleteMessage(id)),
});

export default connect(mapStateToProps, mapDispatchToProps)(Snackbar);
