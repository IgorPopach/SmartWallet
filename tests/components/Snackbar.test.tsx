import React from 'react';
import { shallow, ShallowWrapper } from 'enzyme';
import toJson from 'enzyme-to-json';

import Snackbar from './../../src/containers/Snackbar/Snackbar';

type Props = React.ComponentProps<typeof Snackbar>;

describe('<Snackbar />', () => {
    const props: Props = {
        messages: [
            {
                id: '1',
                title: 'Test alert 1',
                text: 'this is test text for Alert',
                type: 'info',
            },
            {
                id: '2',
                title: 'Test alert 2',
                text: 'this is test 2 text for Alert',
                type: 'success',
            },
            {
                id: '3',
                title: 'Test alert 3',
                text: 'this is test 3 text for Alert',
                type: 'warning',
            },
        ],
        closeAction: jest.fn(),
    };
    let wrapper: ShallowWrapper<Props>;
    beforeAll(() => {
        wrapper = shallow(<Snackbar {...props} />);
    });

    it('should have correct layout', () => {
        expect(toJson(wrapper)).toMatchSnapshot();
    });
});
