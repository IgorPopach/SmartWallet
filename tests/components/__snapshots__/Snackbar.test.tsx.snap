// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Snackbar /> should have correct layout 1`] = `
<div
  className="snackbar"
>
  <Alert
    delay={15000}
    key="1"
    message={
      Object {
        "id": "1",
        "text": "this is test text for Alert",
        "title": "Test alert 1",
        "type": "info",
      }
    }
    onClose={[Function]}
  />
  <Alert
    delay={15000}
    key="2"
    message={
      Object {
        "id": "2",
        "text": "this is test 2 text for Alert",
        "title": "Test alert 2",
        "type": "success",
      }
    }
    onClose={[Function]}
  />
  <Alert
    delay={15000}
    key="3"
    message={
      Object {
        "id": "3",
        "text": "this is test 3 text for Alert",
        "title": "Test alert 3",
        "type": "warning",
      }
    }
    onClose={[Function]}
  />
</div>
`;
